src/segment_language.go

```diff
@@ -84,9 +112,12 @@ func (l *language) setVersion() {
 		l.exitCode = exitErr.exitCode
 		return
 	}
+
 	values := findNamedRegexMatch(l.versionRegex, versionInfo)
-	l.exitCode = 0
 	l.version = values["version"]
+	l.major = values["major"]
```

Comment:
is there ever a changelog URL where the need to split this is present?

Reply:
the url to the release notes are quite different for each languages. Some like golang takes only major.minor, others take all and dotnet is also different. So it's more flexible to follow semver and extract all the data we need.